### PR TITLE
use registerRootComponent for bare-expo entry

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- This is the entry we use for newly ejected projects
- remove upper level values in app.json so we can migrate to `app.config.js` to add env variables to replace global.DETOX which doesn't really work
- use `main` for main component instead of `BareExpo`
- Fix pbxproj name for ejecting

### CODE REVIEW

File: apps/bare-expo/babel.config.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ module.exports = function(api) {
   const moduleResolverConfig = {
     alias: {
       '~expo': 'expo',
-      expo: require.resolve('./moduleResolvers/expoResolver.js'),
+      // expo: require.resolve('./moduleResolvers/expoResolver.js'),
```
Comment (tsapeta) [inline apps/bare-expo/babel.config.js:7]: Is this change intentional? On the other hand, I suppose we can already remove `expoResolver.js` 🤔

**Review by tsapeta — APPROVED** [review]

File: apps/bare-expo/babel.config.js (hunk 1 above)
Comment (EvanBacon) [reply apps/bare-expo/babel.config.js:7]: yes it's intentional because without it the tests fail because the entry point doesn't work 2b115a0
